v2/sender.go

```diff
@@ -74,6 +76,8 @@ func (d *Sender) SendMessage(ctx context.Context, mb MessageBody, options ...fun
 		defer cancel()
 	}
 
+	d.mu.RLock()
+	defer d.mu.RUnlock()
```

Comment:
Any reason to unlock only after the select?
can't we move this lock/unlock sequence inside the go func() below, so that the lock is taken just in time and released ASAP?

Reply:
updated to release the lock right after Send()